File: backend/src/models/job.ts
import mongoose from "mongoose";

const Schema = mongoose.Schema

let Job = new Schema({
    title: {
        type: String
    },
    page: {
        type: String
    },
    page_image: {
        type: String
    },
    workplace_type: {
        type: String
    },
    location: {
        type: String
    },
    job_type: {
        type: Number
    },
    level: {
        type: String
    },
    description: {
        type: String
    },
    tech_stack: {
        type: String
    },
    deadline: {
        type: Number
    },
    area: {
        type: String
    },
    date_posted: {
        type: Number
    },
    applicants: [
        {
            user: {
                type: String
            },
        }
    ]
})

export default mongoose.model('Job', Job, 'job')

File: WorkHub/app/src/main/java/com/example/workhub/data/retrofit/models/Job.kt
package com.example.workhub.data.retrofit.models

data class Job(
    val _id: String,
    val title: String,
    val page: String,
    val page_image: String,
    val workplace_type: String,
    val location: String,
    val job_type: Int,
    val level: String,
    val description: String,
    val tech_stack: String,
    val deadline: Long,
    val area: String,
    val date_posted: Long,
    var applicants: List<Applicant>
)

data class Applicant(
    val user: String
)
